### PR TITLE
api: update ProxyLogging

### PR DESCRIPTION
**Pull request by zirain**
cc @arkodg

### CODE REVIEW

File: api/config/v1alpha1/envoyproxy_types.go [hunk 1]
```diff
@@ -156,7 +150,7 @@ const (
 
 // LogLevel defines a log level for system logs. This type is not implemented until
 // https://github.com/envoyproxy/gateway/issues/280 is fixed.
-// +kubebuilder:validation:Enum=debug;info;error
+// +kubebuilder:validation:Enum=debug;info;warning;error
```
[the remark this answers is not in the document]
Comment (zirain) [reply api/config/v1alpha1/envoyproxy_types.go:153]: sure

File: api/config/v1alpha1/envoyproxy_types.go [hunk 2]
```diff
@@ -105,25 +102,22 @@ type EnvoyProxyKubernetesProvider struct {
...
-// is fixed.
 // +kubebuilder:validation:Enum=system;upstream;http;connection;admin;client;filter;main;router;runtime
 type LogComponent string
 
 const (
-	// LogComponentSystem defines the "system"-wide logging component. When specified,
-	// all other logging components are ignored.
+	// LogComponentSystem defines the "system"-wide logging component.
+	// See more details: https://www.envoyproxy.io/docs/envoy/latest/operations/cli#cmdoption-l
 	LogComponentSystem LogComponent = "system"
```
Comment (arkodg) [inline api/config/v1alpha1/envoyproxy_types.go:121]: I dont see any component level called `system`https://github.com/envoyproxy/envoy/blob/cf1af7660d910c275bee1d7b34218c32c6419198//source/common/common/logger.h

Comment (zirain) [reply api/config/v1alpha1/envoyproxy_types.go:121]: as we talked in WG meeting, will change this to `default`.

**Review by arkodg — APPROVED** [review]
LGTM, thanks !

File: internal/infrastructure/kubernetes/proxy/testdata/deployments/bootstrap.yaml [hunk 3]
```diff
@@ -33,7 +33,7 @@ spec:
             - --service-cluster default
             - --service-node $(ENVOY_POD_NAME)
             - --config-yaml test bootstrap config
-            - --log-level info
+            - --log-level error
```
Comment (arkodg) [inline internal/infrastructure/kubernetes/proxy/testdata/deployments/bootstrap.yaml:36]: shouldnt this be `warn` by default ?

Comment (zirain) [reply internal/infrastructure/kubernetes/proxy/testdata/deployments/bootstrap.yaml:36]: yes, thanks for catching

**Review by qicz — APPROVED** [review]
LGTM

**Review by arkodg — APPROVED** [review]

**Pull request merged**